### PR TITLE
I have implemented the missing JVstPersistence.loadBank()

### PR DESCRIPTION
**Pull request by urish**

### CODE REVIEW

**Comment by mhroth** [comment]
Hi urish, sorry about my delay in replying. I missed the notification e-mail from Github. I will look at this change ASAP and get back to you. Thank you very much for the PR!

File: src/com/synthbot/audioplugin/vst/vst2/JVstPersistence.java [hunk 1]
```diff
@@ -140,11 +146,99 @@ public static void loadPreset(JVstHost2 vst, File file) throws DataFormatExcepti
   }
   
   /**
-   * This method is not yet implemented.
+   * Loads a preset bank from file to the current bank of the plugin.
+   * @param vst  The plugin to load the data into.
+   * @param file  The file to read which contains the preset data.
+   * @author Uri Shaked
```
Comment (mhroth) [inline src/com/synthbot/audioplugin/vst/vst2/JVstPersistence.java]: I'm glad that you added this. Would you please also add some way to get in touch with you, e.g.
`@author Uri Shaked (uri@email.com)` or `@author Uri Shaked`(https://github.com/urish)`. Whatever you feel comfortable with.

File: src/com/synthbot/audioplugin/vst/vst2/JVstPersistence.java [hunk 2]
```diff
@@ -140,11 +146,99 @@ public static void loadPreset(JVstHost2 vst, File file) throws DataFormatExcepti
...
+      isRegularChunk = true;
+    } else if (OPAQUE_BANK_MAGIC.equals(chunkDataType)) {
+      if (!vst.acceptsProgramsAsChunks()) {
+        fxp.close();
+        throw new DataFormatException("File contains opaque data but plugin claims not to accept programs as chunks.");
+      } else {
+        isRegularChunk = false;
+      }
+    } else {
+      throw new DataFormatException("File reports that is contains neither regular nor opqaue chunks.");
```
Comment (mhroth) [inline src/com/synthbot/audioplugin/vst/vst2/JVstPersistence.java]: Add `fxp.close();` above this line.

Comment (mhroth) [inline src/com/synthbot/audioplugin/vst/vst2/JVstPersistence.java]: I believe that there is a similar error (my fault at https://github.com/urish/jvsthost/blob/5679b6840930b7a7921bd11139725a23ba401060/src/com/synthbot/audioplugin/vst/vst2/JVstPersistence.java#L102). Would you please also correct this?

**Comment by urish** [comment]
Okay, I have fixed the bug you found (non closing the input stream in case of error) + added my email to the comment

**Comment by mhroth** [comment]
Looks good to me. I like the way that you used `finally {fxp.close();}`, that's much more elegant than the way that I had done it. Thanks again Uri!

**Pull request merged**